eth/watcher/watcher.go

```diff
@@ -421,9 +421,13 @@ func (w *Watch) fetchLogEvents() {
 		w.fromBlock = maxBlock + 1
 		log.Tracef("added %d logs to queue: %s: next from %d", count, w.name, w.fromBlock)
 	} else {
-		// we didn't find any event between fromBlock and toBlock, so we can fast forward
-		// maybe we should also do this above instead of maxBlock + 1?
-		w.fromBlock = toBlock
+		// we didn't find any event between fromBlock and toBlock, so we can fast forward.
+		// add additional block delay to mitigate consistency issues from query nodes,
+		// may add another parameter later
+		fromBlock := toBlock - w.blkDelay
+		if fromBlock > w.fromBlock {
+			w.fromBlock = fromBlock
```

Comment:
missing code of setting "w.fromBlock = toBlock" before moving it conditionally up to "fromBlock"

Reply:
forget what I said, misread the code